tools/topology/topology2/platform/intel/sdw-amp-generic.conf

```diff
@@ -54,6 +55,19 @@ IncludeByKey.PASSTHROUGH {
 				Object.Widget.alh-copier.1 {
 					stream_name $SDW_SPK_STREAM
 					node_type $ALH_LINK_OUTPUT_CLASS
+					IncludeByKey.SDW_AMP_FMT_24 {
+					"true"	{
+							num_output_audio_formats 1
+							Object.Base.output_audio_format [
+								{
+									out_bit_depth           32
+									out_valid_bit_depth     24
+									out_sample_type		$SAMPLE_TYPE_MSB_INTEGER
```

Comment:
@bardliao Sorry for late follow-up. But I'm a bit confused again, this format "MSB_INTEGER" is not S32_LE (24 valid bits left aligned) but rather S24_LE (24bits right-aligned). So what's the format the audio codec does not support?

Reply:
And answer from @bardliao  that it's specifically the case of valid_bit_depth 32 not supported by some of the codecs.